api/dps/index_test.go

```diff
@@ -0,0 +1,456 @@
+		}
+	})
+
+	t.Run("handles index failures", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetCommitFunc: func(_ context.Context, in *GetCommitRequest, _ ...grpc.CallOption) (*GetCommitResponse, error) {
+					assert.Equal(t, testHeight, in.Height)
+
+					return nil, mocks.DummyError
+				},
+			},
+		}
+
+		_, err := index.Commit(testHeight)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles invalid indexed data", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetCommitFunc: func(_ context.Context, in *GetCommitRequest, _ ...grpc.CallOption) (*GetCommitResponse, error) {
+					assert.Equal(t, testHeight, in.Height)
+
+					return &GetCommitResponse{
+						Height: testHeight,
+						Commit: []byte(`not a commit`),
+					}, nil
+				},
+			},
+		}
+
+		_, err := index.Commit(testHeight)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIndex_Events(t *testing.T) {
+	testHeight := uint64(42)
+	testTypes := []flow.EventType{"deposit", "withdrawal"}
+	testEvents := []flow.Event{
+		{Type: "deposit"},
+		{Type: "withdrawal"},
+	}
+	testEventsB, err := cbor.Marshal(testEvents)
+	require.NoError(t, err)
+
+	t.Run("nominal case", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetEventsFunc: func(_ context.Context, in *GetEventsRequest, _ ...grpc.CallOption) (*GetEventsResponse, error) {
+					assert.Equal(t, testHeight, in.Height)
+					assert.Equal(t, convert.TypesToStrings(testTypes), in.Types)
+
+					return &GetEventsResponse{
+						Height: testHeight,
+						Types:  convert.TypesToStrings(testTypes),
+						Data:   testEventsB,
+					}, nil
+				},
+			},
+		}
+
+		got, err := index.Events(testHeight, testTypes...)
+
+		if assert.NoError(t, err) {
+			assert.Equal(t, testEvents, got)
+		}
+	})
+
+	t.Run("handles index failures", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetEventsFunc: func(_ context.Context, in *GetEventsRequest, _ ...grpc.CallOption) (*GetEventsResponse, error) {
+					assert.Equal(t, testHeight, in.Height)
+					assert.Equal(t, convert.TypesToStrings(testTypes), in.Types)
+
+					return nil, mocks.DummyError
+				},
+			},
+		}
+
+		_, err := index.Events(testHeight, testTypes...)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles invalid indexed data", func(t *testing.T) {
+		index := Index{
+			client: &apiMock{
+				GetEventsFunc: func(_ context.Context, in *GetEventsRequest, _ ...grpc.CallOption) (*GetEventsResponse, error) {
+					assert.Equal(t, testHeight, in.Height)
+					assert.Equal(t, convert.TypesToStrings(testTypes), in.Types)
+
+					return &GetEventsResponse{
+						Height: testHeight,
+						Types:  convert.TypesToStrings(testTypes),
+						Data:   []byte(`invalid data`),
+					}, nil
+				},
+			},
+		}
+
+		_, err := index.Events(testHeight, testTypes...)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIndex_Registers(t *testing.T) {
+	testHeight := uint64(42)
+	path1, _ := ledger.ToPath([]byte("aac513eb1a0457700ac3fa8d292513e1"))
```

Comment:
We might want to `assert`/`require` here just so we catch any errors if something changes.

Reply:
Or we might just want to use `ledger.Path{0xaa, 0xc5, ...}` actually.